src/geometry.rs

```diff
@@ -163,14 +163,39 @@ pub struct Size<T> {
 impl<T> Size<T> {
     /// Applies the function `f` to both the width and height
     ///
-    /// This is used to transform a `Rect<T>` into a `Rect<R>`.
+    /// This is used to transform a `Size<T>` into a `Size<R>`.
     pub fn map<R, F>(self, f: F) -> Size<R>
     where
         F: Fn(T) -> R,
     {
         Size { width: f(self.width), height: f(self.height) }
     }
 
+    /// Applies the function `f` to the width
+    pub fn map_width<F>(self, f: F) -> Size<T>
+    where
+        F: Fn(T) -> T,
+    {
+        Size { width: f(self.width), height: self.height }
+    }
+
+    /// Applies the function `f` to the height
+    pub fn map_height<F>(self, f: F) -> Size<T>
+    where
+        F: Fn(T) -> T,
+    {
+        Size { width: self.width, height: f(self.height) }
+    }
+
+    /// Applies the function `f` to both the width and height
+    /// of this value and another passed value
+    pub fn zip_map<Other, Ret, Func>(self, other: Size<Other>, f: Func) -> Size<Ret>
```

Comment:
Is there a precedent for calling this method "zip"? I haven't seen that before, and from my perspective it would not be intuitive what it would do.

Reply:
Yep, see https://stackoverflow.com/questions/1115563/what-is-zip-functional-programming